File: 프로그래머스/lv2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    answer = 0
    for i in range(len(board)-1):
        for j in range(len(board[0])-1):
            if board[i+1][j+1] ==1:
                board[i+1][j+1] = min(board[i][j],board[i+1][j],board[i][j+1])+1
                answer = max(answer, board[i+1][j+1])
    if len(board[0])<=2:
        answer = 1
    return answer*answer

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.js
function solution(board)
{
    let answer = 0;
    let row = board.length;
    let col = board[0].length;

    if(row < 2 || col < 2) return 1;

    for(let i = 1; i < row; i++) {
        for(let j = 1; j < col; j++) {
            if(board[i][j] !== 0) {
                let min = Math.min(
                    board[i-1][j-1],
                    board[i-1][j],
                    board[i][j-1]
                );
                board[i][j] = min + 1;
            }
            if(answer < board[i][j]) answer = board[i][j];
        }
    }
    return answer * answer;
}